Client/Frontend/Library/Bookmarks/BookmarksPanel.swift

```diff
@@ -151,57 +111,153 @@ class BookmarksPanel: SiteTableViewController, LibraryPanel, CanRemoveQuickActio
         }
     }
 
+    // MARK: - Data
+
     override func reloadData() {
         // Can be called while app backgrounded and the db closed, don't try to reload the data source in this case
         if profile.isShutdown { return }
-        profile.places.getBookmarksTree(rootGUID: bookmarkFolderGUID, recursive: false).uponQueue(.main) { result in
 
+        if bookmarkFolderGUID == BookmarkRoots.RootGUID {
+            setupRootFolderData()
```

Comment:
😍 for the separate setup functions way easier to understand the reload function now